mininet/node.py

```diff
@@ -838,6 +838,38 @@ def init( cls ):
         mountCgroups()
         cls.inited = True
 
+class XIAHost( Host ):
+
+    "XIA host"
+
+    def __init__( self, name, **kwargs ):
+        ""
+       	Node.__init__( self, name, **kwargs )
+	self.xids = {}        
+
+    def config( self, hid=None, **params ):
```

Comment:
Do we need to have a parameter `hid` here? In most cases we won't be calling this directly, we'll be using `configDefault()` from within the `build()` method, which invokes this method using:

    self.config( **self.params )

And therefore we won't get a variable `hid`. However, `hid` should already be stored in `params`.

Reply:
Yes, so the overridden `config()` which is called by `configDefault()` now performs all the additional configurations(other than mac, ip and loopback).  So the `hid=None` catches the hid inputs and the remaining are used to invoke Node.config().